spec.md

```diff
@@ -283,6 +293,69 @@ intmodule MyIntrinsicModule_xhello_y64 :
   parameter y = 42
 ```
 
+The types of intrinsic module parameters may only be literal integers or
+string literals.
+
+A parameter that is a string-encoded integer literal is treated as a string
+literal.
+
+# Literals
+
+FIRRTL has both integer, string, and raw string literals.
+
+An integer literal is a signed or unsigned decimal integer.  The following are
+examples of integer literals:
+
+``` firrtl
+42
+-9000
+```
+
+A string literal is a sequence of characters with a leading `"`{.firrtl} and a
+trailing `"`{.firrtl}.  The following is an example of a string literal:
+
+``` firrtl
+"hello"
+```
+
+A raw string literal is a sequence of characters with a leading `'`{.firrtl} and
+a trailing `'`{.firrtl}.  The following is an example of a raw string literal:
+
+``` firrtl
+'world'
+```
+
+## String-encoded Integer Literal
+
+A string-encoded integer literal is a special string literal with one of the
+following leading characters to indicate the numerical encoding:
+
+- `b`{.firrtl} -- for representing binary numbers
+- `o`{.firrtl} -- for representing octal numbers
+- `h`{.firrtl} -- for representing hexadecimal numbers
+
+Signed string-encoded integer literals have their sign following the leading
+encoding character.
+
+The following string-encoded integer literals all have the value `42`:
+
+``` firrtl
+"b101010"
+"o52"
+"h2a"
+```
+
+The following string-encoded integer literals all have the value `-42`:
+
+``` firrtl
+"b-101010"
```

Comment:
I'd rather it be -b101010.  Then at least it looks like negation of an integer literal rather than an identifier minus an interger literal.

Is this supported now or is this being added by this?

Reply:
I agree. This is supported now, though. 🥲